webapp/src/app/shared/widgets/character-widget/character-widget.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit , Input} from '@angular/core';
+import {AbstractWidget} from '../abstract-widget';
+
+@Component({
+  selector: 'app-character-widget',
+  templateUrl: './character-widget.component.html',
+  styleUrls: ['./character-widget.component.scss', '../widget.scss']
+})
+export class CharacterWidgetComponent extends AbstractWidget implements OnInit {
+
+  /** @member {boolean} nested */
+  @Input() nested: boolean = false;
+  constructor() { 
+    super();
+  }
+
+  ngOnInit() {
+    super.ngOnInit();
+  }
+
+  /**
+   * Returns all known character names. 
+   * @todo Implement class to manage character names.
+   * @todo Remove hardcoded example output.
+   * @returns {string[]} Character names in the format of "typeOfCharacter.characterAlias"
+   */
+  getAllNames(): string[] {
+    return ["main.lea"];
```

Comment:
Returning a new array on each call causes a lot of updates. Consider storing this in a field instead. The value of the array should be stored in a json in the asset folder.